app/controllers/deploys_controller.rb

```diff
@@ -49,6 +49,58 @@ def recent
     end
   end
 
+  # Takes the same params that are used by the client side filtering
+  # on the recent deploys pages.
+  # Returrns a paginated json object or CSV of deploys that people are
+  # interested in rather than doing client side filtering.
+  # params:
+  #   * deployer (name of the user that started the job(s), this is a fuzzy match
+  #   * project_name (name of the project)
+  #   * production (boolean, is this in proudction or not)
+  #   * status (what is the status of this job failed|running| etc)
+
+  def search
+    if (params[:status] && !Job.valid_status?(params[:status]))
+      render json: { errors: "invalid status given" }, status: 400
+      return
+    end
+    if params[:deployer].present?
+      users = User.where("name LIKE ?", "%#{ActiveRecord::Base.send(:sanitize_sql_like, params[:deployer])}%").pluck(:id)
+    end
+
+    if params[:project_name].present?
+      projects = Project.where(name: params[:project_name]).pluck(:id)
+    end
+
+    if users || params[:status]
+      jobs = Job
+      jobs = jobs.where(user: users) if users
+      jobs = jobs.where(status: params[:status]) if params[:status]
+    end
+
+    if params[:production] || projects
+      stages = Stage
+      stages = stages.where(project: projects) if projects
+      production = ActiveRecord::Type::Boolean.new.type_cast_from_user(params[:production]) if params[:production]
+      # this will hit the deploy groups as well
```

Comment:
Yup.  production however is seemingly more often related to a deploy group and not the deploy itself